src/SpectralUnmixing.jl

```diff
@@ -393,28 +416,72 @@ end
     prepare_options(library::SpectralLibrary, combination_type::String,
                      num_endmembers::Vector{Int64}, class_idx)
 
-Prepare combinations of endmembers based on the specified combination type.
+Prepare combinations of endmembers based on the specified combination type. 
+
+# Arguments
+- `library::SpectralLibrary`: Endmember library for unmixing.
+- `combination_type::String`: see below.
+- `class_idx`: A collection of indices indicating class memberships for different
+endmembers in `library.spectra`
+-`subsample::Bool`: Whether to create a subsample of all possible combinations instead
+of a full list of combinations. If the number of classes is high (> 7), the full list might
+not be able to be held in memory. 
+-`max_combinations::Int`: The desired number of combinations to obtain.
+-`seed_value::Int`: If subsample is True, the random seed to fix the combination subsample.
+
+# Returns
+- A matrix of integers of size [max_combinations] representing the combinations of endmember indices.
 
 - Combination type options:
     - `"class-even"``: Generate all combinations where one endmember is selected from each class.
     - `"all"`: Generate all possible combinations of `num_endmembers` spectra.
 """
 function prepare_options(library::SpectralLibrary, combination_type::String,
-    num_endmembers::Vector{Int64}, class_idx)
+                        num_endmembers::Vector{Int64}, class_idx; 
+                        subsample::Bool = false, 
+                        max_combinations::Int = 100, 
+                        seed_value::Int = 1234)
 
     # Prepare combinations if relevant
     options = []
-    if combination_type == "class-even"
-        options = collect(Iterators.product(class_idx...))[:]
-    elseif combination_type == "all"
-        for num in num_endmembers
-            combo = [c for c in combinations(1:length(library.classes), num)]
-            push!(options, combo...)
-        end
-    else
-        error("Invalid combiation string")
-    end
 
+    # Prepare combinations if relevant
+    if subsample
```

Comment:
@marchett2  - This looks like it will get handled within unmix_pixel - can you help me with the rationale for pulling it up here (I recall there was one, just not seeing it clearly here when reviewing)?  Is it for simulations or something else?  If so, we should remove from unmix_pixel so that it's not redundant.